modules/sfml/antara/gaming/sfml/komodo.intro.scene.cpp

```diff
@@ -14,26 +14,341 @@
  *                                                                            *
  ******************************************************************************/
 
+#include <SFML/Graphics.hpp>
 #include <utility>
+#include <entt/entity/helper.hpp>
+#include <entt/entity/registry.hpp>
+#include <entt/signal/dispatcher.hpp>
+#include <antara/gaming/ecs/component.position.hpp>
+#include "antara/gaming/config/config.game.hpp"
+#include "antara/gaming/ecs/component.layer.hpp"
 #include "antara/gaming/sfml/komodo.intro.scene.hpp"
+#include "antara/gaming/sfml/component.drawable.hpp"
+#include "antara/gaming/sfml/component.audio.hpp"
+#include "antara/gaming/sfml/event.play.sound.hpp"
+
+// Utility
+static const float DEG2RAD = 0.0174533f;
+
+static bool ease(float* val, const float targetVal, const float rate, const float dt) {
+    if(*val == targetVal) return true;
+
+    if(std::abs(*val - targetVal) < 0.0001f){
+        *val = targetVal;
+        return true;
+    }
+
+    float amount = (targetVal - *val) * rate * dt;
+    if((*val < targetVal && *val + amount > targetVal) ||
+       (*val > targetVal && *val + amount < targetVal)) *val = targetVal;
+    else *val += amount;
+
+    return false;
+}
 
 namespace antara::gaming::sfml
 {
+    intro_scene::animation::animation(float start_time, std::function<bool(float)> animation) :
+        start_time(start_time),
+        animate(std::move(animation)),
+        done(false)
+    {
+
+    }
+
+    void intro_scene::animation::update(float dt) {
+        done = animate(dt);
+    }
+
+    bool intro_scene::animation::is_done() {
+        return done;
+    }
+
     intro_scene::intro_scene(entt::registry &entity_registry, entt::dispatcher &dispatcher,
                              intro_scene::on_finish_functor on_finish_functor) noexcept : base_scene(entity_registry,
                                                                                                      dispatcher),
                                                                                           on_finish_functor_(std::move(
                                                                                                   on_finish_functor))
     {
+        // Load assets
+        load_sprite("logo");
+        load_sprite("name");
+        load_sound("intro1");
+        load_sound("intro2");
+        load_foreground();
+        load_background();
```

Comment:
We can move this in a private member functions `load_entities()` or `create a factory` that will do the job